pyproject.toml

```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "poetry"
-version = "0.12.8"
+version = "attr: setuptools_scm.get_version"
```

Comment:
would there be a way to enable setuptools_scm to provide the version more directly?

Reply:
Currently, version arg is read unconditionally + I didn't notice any hooks in poetry

Reply:
Of course calling setuptools_scm could be hardcoded in python source, but that's a question about flexibility

Reply:
What is meant by this question?  It looks to me like the setuptools_scm library is directly responsible for providing the version in this example.

Reply:
@kevinkjt2000 This example only shows that users can refer any callable they want. setuptools_scm has been chosen as a common example but any function will work. That's flexibility.
OTOH @RonnyPfannschmidt seems to want smth like `use_version scm = True` to work without having to refer to the function explicitly.

Reply:
I may have a lack of understanding here, but to `trigger the integration and the file finder via the setuptools plugin system` is that not what the get_version function does?

Reply:
@kevinkjt2000 setuptools has entry_point-based hook system. Other packages can specify their entry points with those name to "register" their hook implementations and setuptools then automatically knows it can use them. Example: https://github.com/pypa/setuptools_scm/blob/master/setup.py#L69